conans/client/conan_api.py

```diff
@@ -428,11 +428,13 @@ def config_get(self, item):
     def config_set(self, item, value):
         config_parser = ConanClientConfigParser(self._client_cache.conan_conf_path)
         config_parser.set_item(item, value)
+        self._client_cache = get_client_cache(get_conan_user_home(), self._user_io.out)
```

Comment:
I think I'd prefer something in the cache that would invalidate it, instead of creating a new instance again, and having to re-evaluate ``get_conan_user_home()``.
Maybe ``storage_folder`` is really unused and could be simplified?